### PR TITLE
[WIP] Api refactor

### PR DESCRIPTION
**Pull request by Kaik**
Refactor classes in Api folder #293 

To do list.
- [x] AccountApi 
  - [x] getAll - link container
- [x] AdminApi 
  - [x] getLinks - link container
- [x] BlockApi
  - [x] getLastPosts - ~~Post repository~~ need to be moved somewhere else as it uses various services which should not be used in repository
  - [x] getTopForums - ~~Forum repository~~ same as above
  - [x] getTopPosters - Post repository/ForumUsers repository - this can be moved to repository

At the moment BlockApi is moved to BlockHelper service but it might be replaced compleatly by dedicated services for Posts Forums and Posters
- [ ] CronApi
  - [x] mail - service
  - [x] mailcronecho - service
- [ ] FavoritesApi
  - [x] getStatus - ?
  - [x] isFavorite - FavoritesEntity
- [x] ForumApi
  - [x] getParents
  - [x] getAllChildren 
  - [x] getNode
  - [x] getForumsByPermissions moved to Security/Permission
  - [x] isSubscribed 
  - [x] subscribe
  - [x] unsubscribe
  - [x] getSubscriptions
  - [x] modify
- [x] ModeratorsApi
  - [x] get
- [ ] NotifyApi
  - [x] emailSubscribers
  - [x] notify_moderator
  - [x] email
- [ ] ParseTagsApi
  - [x] transform
  - [x] encode_quote
  - [x] encode_code
- [x] PermissionsApi
  - [x] get
  - [x] canSee
  - [x] canRead
  - [x] canWrite
  - [x] canModerate
  - [x] canAdministrate
  - [x] checkPermission
  - [x] filterForumArrayByPermission
  - [x] filterForumChildrenByPermission
- [ ] PostApi
  - [x] getLatest
  - [x] search
  - [x] move
  - [x] checkMessageLength
- [x] PrefsApi
  - [x] getLinks
- [ ] RankApi
  - [x] getAll
  - [x] save
  - [x] assign
  - [x] getData
  - [x] addImageAndLink
  - [x] dzk_isimagefile
- [ ] SyncApi
  - [x] all
  - [x] forums
  - [x] forum
  - [x] addToParentForumCount
  - [x] topics
  - [x] topic
  - [x] posters
  - [x] forumLastPost
  - [x] topicLastPost
- [x] TopicApi
  - [x] changeStatus
  - [x] subscribe
  - [x] unsubscribe
  - [x] getSubscriptions
  - [x] getIdByReference
  - [x] email
  - [x] delete
  - [x] move
  - [x] split
  - [x] join
- [ ] UserApi
  - [x] countstats
  - [x] countEntity
  - [x] setcookies
  - [x] get_viewip_data
  - [x] getTopicPage
  - [x] insterrss
  - [x] isSpam
  - [x] useragentIsBot
  - [x] dskVarPrepHTMLDisplay
  - [x] dzkstriptags
  - [x] getUsersByFragments
  - [x] truncate
- [x] UserDataApi
  - [x] getUserOnlineStatus

This will be very long one... 

...

### CODE REVIEW

**Comment by Kaik** [comment]
Ok I'm back! Last two months were crazy so much traveling and stuff... Now I just need to refresh my testing/working environment's and will start committing.

**Comment by Kaik** [comment]
Ok about  65% maybe 70% is done and I mean whole Dizkus not only Api refactor controllers, blocks, templates, forms... Of course, it is not usable at the moment I hope to finish it, well at least making it usable before New Year...

File: Form/Type/Topic/ReplyType.php [hunk 1]
```diff
@@ -26,7 +27,14 @@ public function __construct($loggedIn) {
     public function buildForm(FormBuilderInterface $builder, array $options) {
 
         $builder->add('message', 'textarea', [
```
Comment (craigh) [inline Form/Type/Topic/ReplyType.php]: would be wise to convert `'textarea'` to `Symfony\Component\Form\Extension\Core\Type\TextAreaType` and all similar items. this makes them Symfony 3.0 compatible and able to run on Core-2.0 sooner

Comment (Kaik) [inline Form/Type/Topic/ReplyType.php]: Ok

**Comment by craigh** [comment]
nice fixes 👍

**Comment by craigh** [comment]
btw - you could have made StyleCI fix some of this FOR you. When you have conflicts from StyleCI click on the link to show the conflicts, then click the login button at the top right. Then click 'create Fix PR' and it will fix them! (you have to merge the PR) and YES you can merge a PR into a PR ;-)

**Comment by Kaik** [comment]
Hmm, I will try this next time... I was checking what I do wrong so I can keep eye on it now - like blank spaces, some code style mistakes. Now I know what to change in Netbeans so it will help me... I know PhpStorm do it by default but I'm already so addicted to Netbeans that changing IDE now would basically ruin all my plans :)

**Comment by craigh** [comment]
you should merge this now and continue finishing fixes in main branch. you have commit rights.

**Comment by Kaik** [comment]
I will finish few things with topic controller and will do merge then. I wonder maybe I should do squash and merge?

**Comment by craigh** [comment]
> I wonder maybe I should do squash and merge?

up to you. I've kind of stopped doing that unless there are a whole lot of 'correction' type commits within the PR (which you _do_ have many of). In the future I think this will become more important (in the core) when we are moving to large **feature** PRs

but it is your choice.

File: Controller/TopicController.php [hunk 2]
```diff
@@ -338,73 +338,38 @@ public function deletetopicAction(Request $request, $topic)
         $hook = new ValidationHook(new ValidationProviders());
         $hookvalidators = $this->get('hook_dispatcher')->dispatch('dizkus.ui_hooks.topic.validate_delete', $hook)->getValidators();
         if ($hookvalidators->hasErrors()) {
-            //return $this->view->registerError($this->__('Error! Hooked content does not validate.'));
+            $request->getSession()->getFlashBag()->add('error', $this->__('Error! Hooked content does not validate.'));
```
Comment (craigh) [inline Controller/TopicController.php]: try `$this->addFlash()`

Comment (Kaik) [inline Controller/TopicController.php]: Ok I will replace all of those with addFlash()

**Pull request merged**